collections/nemo_nlp/nemo_nlp/data/data_layers.py

```diff
@@ -414,3 +416,119 @@ def dataset(self):
     @property
     def data_iterator(self):
         return self._dataloader
+
+
+class GlueDataLayerClassification(TextDataLayer):
+    """
+    Creates the data layer to use for the GLUE classification tasks,
+    more details here: https://gluebenchmark.com/tasks
+
+    All the data processing is done in GLUEDataset.
+
+    Args:
+        dataset_type (GLUEDataset):
+                the dataset that needs to be converted to DataLayerNM
+    """
+
+    @staticmethod
+    def create_ports():
+        output_ports = {
+            "input_ids": NeuralType({
+                0: AxisType(BatchTag),
+                1: AxisType(TimeTag)
+            }),
+            "input_type_ids": NeuralType({
+                0: AxisType(BatchTag),
+                1: AxisType(TimeTag)
+            }),
+            "input_mask": NeuralType({
+                0: AxisType(BatchTag),
+                1: AxisType(TimeTag)
+            }),
+            "labels": NeuralType({
+                0: AxisType(CategoricalTag),
+            }),
+        }
+        return {}, output_ports
+
+    def __init__(self,
+                 data_dir,
+                 tokenizer,
+                 max_seq_length,
+                 processor,
+                 evaluate=False,
+                 token_params={},
+                 num_samples=-1,
+                 shuffle=False,
+                 batch_size=64,
+                 dataset_type=GLUEDataset,
+                 **kwargs):
+
+        kwargs['batch_size'] = batch_size
+        dataset_params = {'data_dir': data_dir,
+                          'output_mode': 'classification',
+                          'processor': processor,
+                          'evaluate': evaluate,
+                          'token_params': token_params,
+                          'tokenizer': tokenizer,
+                          'max_seq_length': max_seq_length}
+
+        super().__init__(dataset_type, dataset_params, **kwargs)
+
+
+class GlueDataLayerRegression(TextDataLayer):
```

Comment:
we can leave it as it is for now but i think separating GlueDataLayerClassification and GlueDataLayerRegression only makes me more confusing to users.